MvvmCross-Forms/MvvmCross.Forms.Droid/Views/MvxFormsAndroidViewPresenter.cs

```diff
@@ -0,0 +1,272 @@
+		{
+            return false;
+		}
+
+        protected virtual void ShowContentPage(
+            Type view,
+            MvxContentPagePresentationAttribute attribute,
+            MvxViewModelRequest request)
+        {
+            var page = CreatePage(view, request);
+
+            if(attribute.WrapInNavigationPage && (FormsApplication.MainPage == null || FormsApplication.MainPage.GetType() != typeof(MvxNavigationPage)))
+            {
+                FormsApplication.MainPage = new MvxNavigationPage(page);
+            }
+            else if(attribute.WrapInNavigationPage && FormsApplication.MainPage is MvxNavigationPage navigationPage)
+            {
+                navigationPage.PushAsync(page, attribute.Animated);
+            }
+            else
+            {
+                FormsApplication.MainPage = page;
+            }
+        }
+
+        protected virtual bool CloseContentPage(IMvxViewModel viewModel, MvxContentPagePresentationAttribute attribute)
+        {
+            return true;
+        }
+
+        protected virtual void ShowMasterDetailPage(
+            Type view,
+            MvxMasterDetailPagePresentationAttribute attribute,
+            MvxViewModelRequest request)
+        {
+            var page = CreatePage(view, request) as MasterDetailPage;
+            FormsApplication.MainPage = page;
+        }
+
+        protected virtual bool CloseMasterDetailPage(IMvxViewModel viewModel, MvxMasterDetailPagePresentationAttribute attribute)
+        {
+            return true;
+        }
+
+        protected virtual void ShowModal(
+            Type view,
+            MvxModalPresentationAttribute attribute,
+            MvxViewModelRequest request)
+        {
+            var page = CreatePage(view, request);
+
+            if (FormsApplication.MainPage is MvxNavigationPage navigationPage)
+                navigationPage.CurrentPage.Navigation.PushModalAsync(page);
+        }
+
+        protected virtual bool CloseModal(IMvxViewModel viewModel, MvxModalPresentationAttribute attribute)
+        {
+            if (FormsApplication.MainPage is MvxNavigationPage navigationPage)
+                navigationPage.CurrentPage.Navigation.PopModalAsync();
+            return true;
+        }
+
+        protected virtual void ShowNavigationPage(
+            Type view,
+            MvxNavigationPagePresentationAttribute attribute,
+            MvxViewModelRequest request)
+        {
+            var page = CreatePage(view, request);
+            FormsApplication.MainPage = page;
+        }
+
+        protected virtual bool CloseNavigationPage(IMvxViewModel viewModel, MvxNavigationPagePresentationAttribute attribute)
+        {
+            return true;
+        }
+
+        protected virtual void ShowTabbedPage(
+            Type view,
+            MvxTabbedPagePresentationAttribute attribute,
+            MvxViewModelRequest request)
+        {
```

Comment:
Just trying to wrap my head around the presenter implementations.... So for the sake of my mind, you would need to make a MvxTabbedPage which implements the IMvxPage, in order for CreatePage to actually set the VM?

Reply:
`MvxTabbedPage` already implements `IMvxPage` so it will set the VM.